dasharo-security/tpm2-commands.robot

```diff
@@ -225,6 +225,39 @@ TPMCMD011.001 Performing HMAC operation on the file (Ubuntu 22.04)
     Should Contain    ${out1}    hmac.out
     Should Not Contain    ${out2}    hmac.out
 
+TPMCMD012.001 Encrypt and Decrypt non-rootfs partition (Ubuntu 22.04)
+    [Documentation]    Test encrypting and decrypting non-rootfs partition using
+    ...    TPM.
+    # Check:
+    Check If Package Is Installed    cryptsetup
+    # Prepare:
+    Execute Command In Terminal    fallocate -l 20MB test-partition
+    # Create sealing object:
+    Execute Linux Tpm2 Tools Command    tpm2_createprimary -Q -C o -c prim.ctx
+    Execute Command In Terminal    dd if=/dev/urandom bs=1 count=32 status=none > key
+    Execute Linux Tpm2 Tools Command    cat key | tpm2_create -Q -g sha256 -u seal.pub -r seal.priv -i- -C prim.ctx
+    Execute Linux Tpm2 Tools Command    tpm2_load -Q -C prim.ctx -u seal.pub -r seal.priv -n seal.name -c seal.ctx
+    Execute Linux Tpm2 Tools Command    tpm2_evictcontrol -C o -c seal.ctx 0x81010001
+    Execute Linux Tpm2 Tools Command    tpm2_unseal -Q -c 0x81010001 > key
+    # Encrypt the partition
+    Execute Command In Terminal    cryptsetup luksFormat -q --key-file=key ./test-partition
+    Execute Command In Terminal    cryptsetup luksOpen --key-file=key ./test-partition test-partition
+    # Test by adding a file:
+    Execute Command In Terminal    mkfs.ext4 /dev/mapper/test-partition
+    Execute Command In Terminal    mount /dev/mapper/test-partiton /mnt
+    Execute Command In Terminal    touch /mnt/hello-world
+    Execute Command In Terminal    umount /mnt
+    Execute Command In Terminal    cryptsetup luksClose test-partition
+    Execute Command In Terminal    cryptsetup luksOpen ./test-partition --key-file=key test-partition
+    Execute Command In Terminal    mount /dev/mapper/test-partition /mnt
+    Execute Command In Terminal    ls /mnt | grep hello-world
```

Comment:
@DaniilKl   Execute Command In Terminal doesn't check if command finished successfully, you should probably save returned output and check that it contains `hello-world`

Reply:
@TomaszAIR Note that Ansible support have not been merget yet, so we do not have Ansible support on this brunch.

Reply:
@DaniilKl ansible support is added now into `encrypted-rootfs-release` branch. You can use it by rebasing this work.

Reply:
@m-iwanicki From now on it is checked by KW `Should Contain`, amended here e99207d1747d267dcdee7e33aff10d5bb1955fd7.

@TomaszAIR Added ansible support for this test here da7c1090fab45e49b61f0f47551256f579cffc19.